project-koku/koku
# Uncomment partitioned tables for purge by partition drop

Fixes [COST-2200](https://issues.redhat.com/browse/COST-2200)

Changes proposed in this PR:
* This change includes the following tables for purge-by-partition-drop processing that should have been included on the first PR
    * reporting_ocpawscostlineitem_daily_summary_p
    * reporting_ocpawscostlineitem_project_daily_summary_p
    * reporting_ocpazurecostlineitem_daily_summary_p
    * reporting_ocpazurecostlineitem_project_daily_summary_p

Testing Instructions:
1. Load multiple months of data that will include ocp on aws/azure
2. Fire up the django shell `python manage.py shell`
3. `import datetime`
4. `from masu.processor.aws.aws_report_db_cleaner import AWSReportDBCleaner`
5. `from masu.processor.azure.azure_report_db_cleaner import AzureReportDBCleaner`
6. `from koku.database import get_model`
7. `from django.db import connection`
8. `connection.set_schema("acct10001")`
9. `target_table_names = ["reporting_ocpawscostlineitem_daily_summary", "reporting_ocpawscostlineitem_project_daily_summary", "reporting_ocpazurecostlineitem_daily_summary", "reporting_ocpazurecostlineitem_project_daily_summary"]`
10. `ParTables = get_model("partitioned_tables")`
11. `aws_cleaner = AWSReportDBCleaner("acct10001")`
12. `azure_leaner = AzureReportDBCleaner("acct10001")`
13. `expire_date = datetime.date(<year>, <month>, <last-day-of-month>)`
14. `partitions_before = list(ParTables.objects.filter(partition_of_table_name__in=target_table_names, partition_parameters__default=False).values("schema_name", "table_name").order_by("schema_name", "table_name"))`
15. `aws_res = aws_cleaner.purge_expired_report_data_by_date(expire_date)`
16. `azure_res = azure_cleaner.purge_expired_report_data_by_date(expire_date)`
17. `connection.set_schema("acct10001")`
18. `partitions_after = list(ParTables.objects.filter(partition_of_table_name__in=target_table_names, partition_parameters__default=False).values("schema_name", "table_name").order_by("schema_name", "table_name"))`
19. If you compare `partitions_before` and `partitions_after` you will see partitions removed for the year and month <= the year and month of the `expire_date`

---
Additional Context:

Smokes:

```
16:28:44 =========================== short test summary info ============================
16:28:44 FAILED tests/rest_api/v1/test_resource_type_endpoints.py::test_api_gcp_product_search
16:28:44 FAILED tests/rest_api/v1/test_resource_type_endpoints.py::test_api_gcp_projects_compare_data
16:28:44 FAILED tests/rest_api/v1/test_source.py::test_api_gcp_source_raw_calc - Asser...
16:28:44 FAILED tests/rest_api/v1/test_source.py::test_api_ocp_source_crud_upload_service
16:28:44 = 4 failed, 7011 passed, 847 skipped, 1145 deselected, 2 warnings in 3894.23s (1:04:54) =
```

[Console Log](https://ci.ext.devshift.net/job/project-koku-koku-pr-check/3871/console)
